File: complete/src/main/java/hello/GreetingHandler.java
package hello;

import org.springframework.http.MediaType;
import org.springframework.stereotype.Component;
import org.springframework.web.reactive.function.BodyInserters;
import org.springframework.web.reactive.function.server.ServerRequest;
import org.springframework.web.reactive.function.server.ServerResponse;

import reactor.core.publisher.Mono;

@Component
public class GreetingHandler {

	public Mono<ServerResponse> hello(ServerRequest request) {
		return ServerResponse.ok().contentType(MediaType.TEXT_PLAIN)
			.body(BodyInserters.fromValue("Hello, Spring!"));
	}
}

File: complete-kotlin/src/main/kotlin/com/example/reactivewebservice/GreetingHandler.kt
package com.example.reactivewebservice

import org.springframework.http.MediaType
import org.springframework.stereotype.Component
import org.springframework.web.reactive.function.server.ServerRequest
import org.springframework.web.reactive.function.server.ServerResponse
import reactor.core.publisher.Mono

@Component
class GreetingHandler {

  fun hello(request: ServerRequest): Mono<ServerResponse> =
    ServerResponse.ok()
      .contentType(MediaType.APPLICATION_JSON)
      .bodyValue(Greeting("Hello, Spring!"))
}
